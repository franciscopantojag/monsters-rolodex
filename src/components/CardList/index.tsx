import { useEffect, useState } from 'react';
import { Monster } from 'src/types';
import { filterMonsters } from '../../utils';
import Card from '../Card';
import './styles.css';

interface Props {
  searchField: string;
}

const CardList: React.FC<Props> = ({ searchField = '' }) => {
  const [monsters, setMonsters] = useState<Monster[]>([]);
  const [filteredMonsters, setFilteredMonsters] = useState<Monster[]>([]);

  useEffect(() => {
    const main = async () => {
      const monsters: Monster[] = await (await fetch('https://jsonplaceholder.typicode.com/users')).json();
      setMonsters(() => monsters);
    };
    main();
  }, []);

  useEffect(() => {
    setFilteredMonsters(() => filterMonsters(monsters, searchField));
  }, [searchField, monsters]);

  return (
    <div className="card-list">
      {filteredMonsters.map((monster) => (
        <Card key={monster.id} monster={monster} />
      ))}
    </div>
  );
};

export default CardList;
