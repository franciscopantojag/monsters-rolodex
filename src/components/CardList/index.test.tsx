import { shallow } from 'enzyme';
import CardList from '.';

it('expect to render Card Component', () => {
  expect(shallow(<CardList searchField="" />)).toMatchSnapshot();
});
