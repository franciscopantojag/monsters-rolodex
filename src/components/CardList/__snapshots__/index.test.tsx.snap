// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card Component 1`] = `
<div
  className="card-list"
/>
`;
